src/main/webapp/app/modeling-submission/modeling-submission.component.ts

```diff
@@ -94,19 +95,16 @@ export class ModelingSubmissionComponent implements OnInit, OnDestroy, Component
                         }
                         this.participation = modelingSubmission.participation;
                         this.modelingExercise = this.participation.exercise as ModelingExercise;
-                        this.problemStatement = this.artemisMarkdown.htmlForMarkdown(this.modelingExercise.problemStatement);
                         if (this.modelingExercise.course) {
                             this.complaintService.getNumberOfAllowedComplaintsInCourse(this.modelingExercise.course.id).subscribe((allowedComplaints: number) => {
                                 this.numberOfAllowedComplaints = allowedComplaints;
                             });
                         }
-                        /**
-                         * set diagramType to class diagram if exercise is null
-                         */
                         if (this.modelingExercise.diagramType == null) {
                             this.modelingExercise.diagramType = DiagramType.ClassDiagram;
                         }
                         this.isActive = this.modelingExercise.dueDate == null || new Date() <= moment(this.modelingExercise.dueDate).toDate();
+                        this.isAfterAssessmentDueDate = !this.modelingExercise.assessmentDueDate || moment().isAfter(this.modelingExercise.assessmentDueDate);
```

Comment:
I am not really happy about the fact that we calculate this in different components over and over again. At first I thought about a getter in the `Exercise` model, but unfortunately this does not work with the current approach. The reason is that we do not have an actual instance of the `ModelingExercise` class here, we just have a plain JavaScript object that has the same properties as defined in the `ModelingExercise` class. At least we assume that it has these properties with the line `this.modelingExercise = this.participation.exercise as ModelingExercise;` above, but we never actually create a new `ModelingExercise` object using its constructor.